### PR TITLE
Move EzButton to Storybook Docs

### PR DESCRIPTION
**Pull request by noranda**
_Note: This branch is based off `nb/storybook-docs-ezblankstate` and should therefore be merged after https://github.com/ezcater/recipe/pull/970._

## What did we change?
- [move EzButton to storybook docs](https://github.com/ezcater/recipe/commit/6c32fc0237bad2a6998cd4f573384b7f694af0cd)

## Why are we doing this?

This is part of an effort to switch our doc site from Gatbsy to Storybook Docs.

_Note: This is an internal documentation change that doesn't affect the recipe core library and therefore has no changeset._

## Screenshot(s) / Gif(s):
Screenshot 2023-06-14 at 2 53 49 PM
Screenshot 2023-06-14 at 2 54 10 PM
Screenshot 2023-06-14 at 2 54 31 PM
Screenshot 2023-06-14 at 2 54 51 PM
Screenshot 2023-06-14 at 2 55 12 PM

## Checklist

- [x] Create new file: `src/components/EzComponent/Documentation/EzComponent.mdx` and populate from Gastby `.md` file.
- [x] Create new file for default story: `src/components/EzComponent/Documentation/Stories/Default.stories.tsx`
- [x] Create any remaining stories grouped as needed: `src/components/EzComponent/Documentation/Stories/Example1Stories.stories.tsx`
- [x] Create snapshot: `src/components/EzComponent/Documentation/EzComponent.snapshot.tsx`
- [x] Add any regression tests as a story: `src/components/EzComponent/Documentation/Stories/Regression.stories.tsx`
- [x] Move (or create) `EzComponent.snippets.tsx` into Documentation folder and make sure snippet matches default story
- [x] Delete old Gatsby files:
  - `src/components/EzComponent/EzComponent.md`
  - `src/components/EzComponent/EzComponent.preview.md`
  - `src/components/EzComponent/__tests__/EzComponent.test.md`
- [x] Test all the things:
  - [x] All control props work as expected
  - [x] All "Show Code" displays correctly
  - [x] Playroom works for docs and all stories (except regression)
  - [x] All action and interaction tests pass for all stories
  - [x] Regression stories only show up in development mode: `STORYBOOK_INCLUDE_REGRESSION_STORIES=1 yarn storybook`
  - [x] Component library preview and link works
  - [x] Related component links work (if already in Storybook Docs)
  - [x] Add any changesets (only if there are recipe library changes)
  - [x] Git commits are organized
  - [x] `yarn lint` passes
  - [x] `yarn type-check` passes
  - [x] `yarn test` passes
  - [x] All chromatic tests pass/approved

  [Templates](https://github.com/ezcater/recipe/discussions/940)

### CODE REVIEW

**Review by danidewitt — APPROVED** [review]

File: packages/recipe/src/components/EzButton/Documentation/Stories/ButtonSize.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+  title: 'Input/EzButton/Button Size',
+};
+
+export default meta;
+type Story = StoryObj<typeof EzButton>;
+
+const playroomCode = (
+  size: EzButtonMuiProps['size'],
+  variant: EzButtonMuiProps['variant']
+) => dedent`
```
Comment (onocy) [inline packages/recipe/src/components/EzButton/Documentation/Stories/ButtonSize.stories.tsx:20]: 🎓 What's the purpose of dedent here?

**Review by onocy — APPROVED** [review]
🎓 TIL about `.mdx` files and `dedent`. 

How did the the jsx defined in the `.md` files we have before get converted into the actual components?

File: packages/recipe/src/components/EzButton/Documentation/Stories/ButtonSize.stories.tsx (hunk 1 above)
Comment (noranda) [reply packages/recipe/src/components/EzButton/Documentation/Stories/ButtonSize.stories.tsx:20]: [Dedent](https://www.npmjs.com/package/dedent-js) lets us format the code in multi-lines. Really useful for our playroom and source code examples!

**Comment by noranda** [comment]
> 🎓 TIL about `.mdx` files and `dedent`.

Nice!

> How did the the jsx defined in the `.md` files we have before get converted into the actual components?

We use a custom-built tool to be able to use jsx and markdown together, which is pretty complex and we just discovered is leading to issues with Chromatic. Storybook comes with full mdx support so everything just works!

**Pull request merged**